contracts/modules/dispute/policies/UMA/ArbitrationPolicyUMA.sol

```diff
@@ -0,0 +1,272 @@
+
+    /// @notice Executes custom logic on disputing judgement
+    /// @dev Enforced to be only callable by the DisputeModule
+    /// @param disputeId The dispute id
+    /// @param decision The decision of the dispute
+    /// @param data The arbitrary data used to set the dispute judgement
+    function onDisputeJudgement(uint256 disputeId, bool decision, bytes calldata data) external onlyDisputeModule {}
+
+    /// @notice Executes custom logic on disputing cancel
+    /// @dev Enforced to be only callable by the DisputeModule
+    /// @param caller Address of the caller
+    /// @param disputeId The dispute id
+    /// @param data The arbitrary data used to cancel the dispute
+    function onDisputeCancel(address caller, uint256 disputeId, bytes calldata data) external onlyDisputeModule {
+        revert Errors.ArbitrationPolicyUMA__CannotCancel();
+    }
+
+    /// @notice Executes custom logic on resolving dispute
+    /// @dev Enforced to be only callable by the DisputeModule
+    /// @param caller Address of the caller
+    /// @param disputeId The dispute id
+    /// @param data The arbitrary data used to resolve the dispute
+    function onResolveDispute(address caller, uint256 disputeId, bytes calldata data) external onlyDisputeModule {}
+
+    /// @notice Allows the IP that was targeted to dispute the assertion while providing counter evidence
+    /// @param assertionId The identifier of the assertion that was disputed
+    /// @param counterEvidenceHash The hash of the counter evidence
+    function disputeAssertion(bytes32 assertionId, bytes32 counterEvidenceHash) external nonReentrant {
+        if (counterEvidenceHash == bytes32(0)) revert Errors.ArbitrationPolicyUMA__NoCounterEvidence();
+
+        ArbitrationPolicyUMAStorage storage $ = _getArbitrationPolicyUMAStorage();
+        uint256 disputeId = $.assertionIdToDisputeId[assertionId];
+        if (disputeId == 0) revert Errors.ArbitrationPolicyUMA__DisputeNotFound();
+
+        (address targetIpId, , , address arbitrationPolicy, , , , uint256 parentDisputeId) = IDisputeModule(
+            DISPUTE_MODULE
+        ).disputes(disputeId);
+
+        if (arbitrationPolicy != address(this)) revert Errors.ArbitrationPolicyUMA__OnlyDisputePolicyUMA();
+        if (parentDisputeId > 0) revert Errors.ArbitrationPolicyUMA__CannotDisputeAssertionIfTagIsInherited();
+
+        // Check if the address can dispute the assertion depending on the liveness and the elapsed time
+        IOptimisticOracleV3.Assertion memory assertion = OPTIMISTIC_ORACLE_V3.getAssertion(assertionId);
+        uint64 liveness = assertion.expirationTime - assertion.assertionTime;
+        uint64 elapsedTime = uint64(block.timestamp) - assertion.assertionTime;
+        bool inIpOwnerTimeWindow = elapsedTime <= (liveness * $.ipOwnerTimePercent) / MAX_PERCENT;
+        if (inIpOwnerTimeWindow && msg.sender != targetIpId)
+            revert Errors.ArbitrationPolicyUMA__OnlyTargetIpIdCanDisputeWithinTimeWindow();
```

Comment:
Would it better including `liveness` and `caller` in the error message for better debugability?

Reply:
addressed in https://github.com/storyprotocol/protocol-core-v1/pull/263/commits/5d331615607da26f57b7641f74ccafb4f68d38ec